src/se/vidstige/jadb/managers/KeyEvent.java

```diff
@@ -0,0 +1,1383 @@
+    /**
+     * Key code constant: Skip backward media key.
+     */
+    public static final int KEYCODE_MEDIA_SKIP_BACKWARD = 273;
+    /**
+     * Key code constant: Step forward media key.
+     * Steps media forward, one frame at a time.
+     */
+    public static final int KEYCODE_MEDIA_STEP_FORWARD = 274;
+    /**
+     * Key code constant: Step backward media key.
+     * Steps media backward, one frame at a time.
+     */
+    public static final int KEYCODE_MEDIA_STEP_BACKWARD = 275;
+    /**
+     * Key code constant: put device to sleep unless a wakelock is held.
+     */
+    public static final int KEYCODE_SOFT_SLEEP = 276;
+    /**
+     * Key code constant: Cut key.
+     */
+    public static final int KEYCODE_CUT = 277;
+    /**
+     * Key code constant: Copy key.
+     */
+    public static final int KEYCODE_COPY = 278;
+    /**
+     * Key code constant: Paste key.
+     */
+    public static final int KEYCODE_PASTE = 279;
+    /**
+     * Key code constant: Consumed by the system for navigation up
+     */
+    public static final int KEYCODE_SYSTEM_NAVIGATION_UP = 280;
+    /**
+     * Key code constant: Consumed by the system for navigation down
+     */
+    public static final int KEYCODE_SYSTEM_NAVIGATION_DOWN = 281;
+    /**
+     * Key code constant: Consumed by the system for navigation left
+     */
+    public static final int KEYCODE_SYSTEM_NAVIGATION_LEFT = 282;
+    /**
+     * Key code constant: Consumed by the system for navigation right
+     */
+    public static final int KEYCODE_SYSTEM_NAVIGATION_RIGHT = 283;
+
+    private static final int LAST_KEYCODE = KEYCODE_SYSTEM_NAVIGATION_RIGHT;
+
+    private boolean longPress;
+    private JadbDevice device;
+    private LinkedList<String> args;
+
+    private KeyEvent(JadbDevice d){
+        device = d;
+        longPress = false;
+        args = new LinkedList<>();
+    }
+
+    private void longPress() {
+        longPress = true;
+    }
+
+    private void addKey(String key) {
+        args.add(key);
+    }
+
+    public static class Builder {
+
+        private KeyEvent mKeyEvent;
+
+        public Builder(JadbDevice d) {
+            mKeyEvent = new KeyEvent(d);
+        }
+
+        public Builder longPress(){
+            mKeyEvent.longPress();
+            return this;
+        }
+
+        public Builder addKey(int key){
+            mKeyEvent.addKey(String.valueOf(key));
+            return this;
+        }
+
+        public KeyEvent build(){
+            return mKeyEvent;
+        }
+    }
+
+    public void execute() throws IOException, JadbException {
+        if(longPress){
+            args.addFirst("--longpress");
+        }
+        args.addFirst("keyevent");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(device.executeShell("input", args.toArray(new String[args.size()]))));
+        String line;
+        while((line = reader.readLine()) != null){
+            System.out.println(line);
+        }
+    }
+}
```

Comment:
In entire file:
* Why are some warning suppressed?
* Fix whitespace
* Let's keep all the constants in this file and move out the methods. Perhaps to `KeyEvent`?